pkg/shell/active-pages-modal.jsx

```diff
@@ -25,12 +25,11 @@ import { Button } from "@patternfly/react-core/dist/esm/components/Button/index.
 import { Label } from "@patternfly/react-core/dist/esm/components/Label/index.js";
 import { Split, SplitItem } from "@patternfly/react-core/dist/esm/layouts/Split/index.js";
 import { Modal } from "@patternfly/react-core/dist/esm/components/Modal/index.js";
-import { useDialogs } from "dialogs.jsx";
 import { useInit } from "hooks";
 
 const _ = cockpit.gettext;
 
-export const ActivePagesDialog = ({ frames }) => {
+export const ActivePagesDialog = ({ frames, result }) => {
```

Comment:
I don't like this API at all, `result` is a property now. But what is a `result`, its rather unambiguous and inexplicit. `Dialogs.close` was a lot more intuitive. 

As its a promise you can't pass a function `closeDialog => result.resolve()`.

Reply:
We could also pass `resolve()` and `reject()` separately, or give them different names (like `closeDialog`).  It wouldn't change the fundamental nature of what they are...

The specific type of this is defined in the `DialogResult` typedef, and it depends on what kind of data the dialog wants to return.